### PR TITLE
DOCSP-43148 - Refactor AWS authentication

### PR DESCRIPTION
**Pull request by mongoKart**
Cloned from https://github.com/mongodb/docs-csharp/pull/250 after inadvertently closing v3.0 branch

# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43148

### Staging Links

<li><a href=https://deploy-preview-258--mongodb-docs-csharp.netlify.app/fundamentals/authentication>fundamentals/authentication</a></li><li><a href=https://deploy-preview-258--mongodb-docs-csharp.netlify.app/upgrade/v3>upgrade/v3</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/authentication.txt [hunk 1]
```diff
@@ -153,12 +153,24 @@ MONGODB-AWS
...
-Identity and Access Management (AWS IAM) credentials to authenticate your
-user. You can either specify your credentials explicitly
-or instruct the driver to retrieve them automatically from an external source.
+The ``MONGODB-AWS`` authentication mechanism uses Amazon Web Services
+Identity and Access Management (AWS IAM) credentials to authenticate a user to MongoDB.
 
-The following sections contain code examples that use the following placeholders:
+To use AWS authentication, first perform the following steps:
+
+- Add the ``MongoDB.Driver.Authentication.AWS`` NuGet package to your project.
```
Comment (sanych-sun) [inline source/fundamentals/authentication.txt:161]: Should we provide a link to nuget? https://www.nuget.org/packages/MongoDB.Driver.Authentication.AWS

File: source/fundamentals/authentication.txt [hunk 2]
```diff
@@ -153,12 +153,24 @@ MONGODB-AWS
...
+- Add the following line of code to your application's bootstrap code to register the
+  AWS authentication provider:
+
+  .. code-block:: csharp
+
+     MongoAuthentication.RegisterAwsAuthenticationProvider();
```
Comment (sanych-sun) [inline source/fundamentals/authentication.txt:167]: Replace the code with the following one:
`MongoClientSettings.Extensions.AddAWSAuthentication();`

**Review by sanych-sun — CHANGES_REQUESTED** [review]

File: source/fundamentals/authentication.txt (hunk 2 above)
Comment (mongoKart) [reply source/fundamentals/authentication.txt:167]: That's really weird. I don't know where that line came from.

**Review by sanych-sun — APPROVED** [review]
LGTM

**Pull request merged**